cfme/tests/physical_infrastructure/api/test_physical_server_api.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils import error
+from cfme.utils.rest import assert_response
+from cfme.utils.wait import wait_for
+from manageiq_client.api import APIException
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([LenovoProvider], scope='module')
+]
+
+# The max time that a single test has to perform the entire cycle of action
+# in seconds (5 min)
+TIMEOUT = 300
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+
+@pytest.fixture(scope="module")
+def physical_server(setup_provider_modscope, appliance):
+    physical_server = appliance.rest_api.collections.physical_servers[0]
+    return physical_server
+
+
+@pytest.fixture(scope="module")
+def unexistent_physical_server(setup_provider_modscope, appliance):
+    return appliance.rest_api.get_entity('physical_servers', 999999)
+
+
+def do_post_physical_server(params, appliance, physical_server):
+    """
+    Do post againist a single physical server
+    """
+    appliance.rest_api.post(physical_server.href, **params)
+    return appliance.rest_api.response.ok
+
+
+def perform_physical_server_action(action, appliance, physical_server):
+    """
+    Perform an action againist a single physical server
+    """
+    payload = {
+        "action" : action
+    }
+    return do_post_physical_server(payload, appliance, physical_server)
+
+
+def get_server_attr(attr_name, appliance, provider, physical_server):
+    provider.refresh_provider_relationships()
```

Comment:
QUESTION: Is this really necessary? You are using this function in ``wait_for`` meaning the refresh is repeated over and over again and it's quite time-consuming operation.
If not also decrease ``DELAY`` so we don't need to wait whole minute.

Reply:
Yes, It is necessary. The `refresh` of the MiQ must be done after Physical Provider conclude the entire operation (the points 2, 3 and 4 described on this PR description). The problem is that we can't assume a time to Physical Provider finish the operation, so we can't know when to execute the `refresh` to get the new state of Physical Provider. If we doesn't force the `refresh` and wait the "natural" refresh of the MiQ the tests could last for 30 min (according to my tests). So, even the `refresh` being a onerous task, it is really necessary.